### ISSUE TITLE
Update new widgets to no longer use legacy `Layout` component

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#2022]
The following refactored widgets currently rely on the `Layout` component which should be considered legacy:
* `DashboardAllTrafficWidget` (Analytics)
* `DashboardPopularKeywordsWidget` (Search Console)
* `DashboardPopularPagesWidget` (Analytics)
* `DashboardTopEarningPagesWidget` (AdSense)

Either the `Widget` component should be expanded to include these capabilities or a new simple component should be introduced to cover the following necessary capabilities:
* Display a heading at the top of the component.
* Display a source link at the bottom of the component.
* The top section should always be at the very top of the widget, and the bottom section should always be at the very end of the widget, no matter the widget's height (i.e. there needs to be some kind of `display: flex; flex-direction: column;` around it.
* There should be a light-gray separator between header, main content, and footer.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The following widgets should no longer rely on the legacy `Layout` component, but instead on a new mechanism enabling the same visual appearance:
    * `DashboardAllTrafficWidget` (Analytics)
    * `DashboardPopularKeywordsWidget` (Search Console)
    * `DashboardPopularPagesWidget` (Analytics)
    * `DashboardTopEarningPagesWidget` (AdSense)

## Implementation Brief

* Add optional `header` and `footer` props to `Widget` component which must be React components if passed.
    * `header` would render above the main widget content, `footer` below it.
* Separate the three sections into their own `div`s and add styling using vertical flexbox to widgets similar to how `googlesitekit-layout--fill` has it, which will ensure widgets next to each other will have the same height, and that header and footer always appear at the top/bottom respectively.
* Add necessary styling:
    * Padding is moved to the inner `div`s instead of being on the widget overall.
    * When the `googlesitekit-widget--no-padding` class is present on the widget, only remove padding from the main content.
    * In composite widget layout where widgets themselves should not have padding, only set the "outer" parts of the padding to `0` - e.g. padding between main content and footer should be unaffected.
* Add support for a boolean `noPadding` prop to `Widget`. If passed, the `googlesitekit-widget--no-padding` class is added.
* Add Storybook coverage for widgets in general with various styles.
* Migrate the four widget components from the ACs to rely on `Widget` instead of `Layout`, providing `noPadding` or adding a `SourceLink` component for `footer` as necessary.
* All other widgets that currently pass the `googlesitekit-widget--no-padding` class should instead pass the new `noPadding` prop.
* Merge #2104.

## QA Brief

* With a development build, ensure the **new** widgets All Traffic (Analytics), Most popular keywords (Search Console), Most popular pages (Analytics) and Top Earning Pages (requires Analytics and AdSense connected) still appear as they did before.

## Changelog entry

* Expand `Widget` component to correctly align vertically and support optional header and footer component props.

**Comment by aaemnnosttv** [comment]
@felixarntz IB looks good for the most part, just one thing to potentially iterate on.

I'm not sure if this is the right issue to address it, but I feel like `Widget` consumers shouldn't be responsible for passing layout-specific classes like `googlesitekit-widget--no-padding`. To me, that should be controlled by a prop which `Widget`  handles (e.g. `padding="none"`). The `className` prop on a `Widget` should probably only be used for things like `my-plugin-widget` or something which is less internal.

**Comment by felixarntz** [comment]
@aaemnnosttv Good point, we can improve that as part of this issue. Since we only have a `boolean` use-case for this currently, I'd prefer to keep it simple and only support a `no-padding` boolean prop. I've updated the IB.

**Comment by aaemnnosttv** [comment]
IB ✅  with one exception – `no-padding` should be `noPadding` (like you have in the PR 😄 ).

I'll update this in the IB since you're already aware of it and move it forward 👍

**Comment by aaemnnosttv** [comment]
## QA ❔ 

Each component no longer uses `Layout` inside its main widget
* ☑️ `DashboardAllTrafficWidget` (Analytics)
* ☑️ `DashboardPopularKeywordsWidget` (Search Console)
* ☑️ `DashboardPopularPagesWidget` (Analytics)
* ☑️ `DashboardTopEarningPagesWidget` (AdSense)

@felixarntz one thing which is not addressed here which perhaps should be is the other rendered states which don't render using the `Widget` component (`PreviewTable` in particular). This is related to but also somewhat conflicting with #2013 which is currently defined to use `Layout` in its IB even though the issue applies to both legacy and widget-based components. Of course legacy components need to continue using `Layout` but perhaps we need a generic `WidgetSkeleton` component to use for rendering the preview for new widgets? Otherwise these will all have the same problem as 2013 where the loading state is missing the wrapping elements which won't appear until everything is loaded.

**Comment by felixarntz** [comment]
@aaemnnosttv 
> one thing which is not addressed here which perhaps should be is the other rendered states which don't render using the `Widget` component

I was thinking about that as well, I think we can work on #2013 as it is, and then have a separate issue to revisit all the other cases. Even for the blue CTA boxes, we currently have somewhat inconsistent UI where in some cases the box is wrapped in the white widget container while in other cases it isn't (this applies to both the new widgets and the legacy widgets). I think we should, after #2013, create an issue about wrapping all widget output into a `Widget` component (i.e. all the ones with `wrapWidget: false` need to add the `Widget` wrapper around all alternative states too).

**Comment by aaemnnosttv** [comment]
Sounds good, QA ✅
